core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -591,9 +697,14 @@ class AuthorizerIntegrationTest extends BaseRequestTest {
     val errorCode = RequestKeyToErrorCode(apiKey).asInstanceOf[(AbstractRequestResponse) => Short](response)
 
     val possibleErrorCodes = resources.flatMap { resourceType =>
-      if(resourceType == Topic)
-        // When completely unauthorized topic resources may return an INVALID_TOPIC_EXCEPTION to prevent leaking topic names
-        Seq(resourceType.errorCode, Errors.INVALID_TOPIC_EXCEPTION.code())
+      if (resourceType == Topic)
+        if (apiKey == ApiKeys.PRODUCE || apiKey == ApiKeys.FETCH ) {
+          //Only allowing TOPIC_AUTHORIZATION_FAILED as an error code
+          Seq(resourceType.errorCode)
```

Comment:
I think this is not right. Both fetch and produce should also return `UNKNOWN_TOPIC_OR_PARTITION` if given a topic that the user doesn't have DESCRIBE permission for, right? It seems like neither checks if the user has DESCRIBE permission, which seems to defeat the point of this PR.